hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/WritableName.java

```diff
@@ -79,20 +79,42 @@ public static synchronized String getName(Class<?> writableClass) {
     return writableClass.getName();
   }
 
+  /**
+   * Return the class for a name. Requires the class for name to extend Writable.
+   * See {@link #getClass(String, Configuration, boolean)} if class doesn't extend Writable.
+   * Default is {@link Class#forName(String)}.
+   *
+   * @param name input name.
+   * @param conf input configuration.
+   * @return class for a name.
+   * @throws IOException raised on errors performing I/O.
+   */
+  public static synchronized Class<?> getClass(String name, Configuration conf)
+      throws IOException {
+    return getClass(name, conf, true);
+  }
+
   /**
    * Return the class for a name.
    * Default is {@link Class#forName(String)}.
    *
    * @param name input name.
    * @param conf input configuration.
+   * @param requireWritable if true, require the class for name to extend Writable
    * @return class for a name.
    * @throws IOException raised on errors performing I/O.
    */
-  public static synchronized Class<?> getClass(String name, Configuration conf
-                                            ) throws IOException {
+  public static synchronized Class<?> getClass(String name, Configuration conf,
+      boolean requireWritable) throws IOException {
```

Comment:
I'd propose that we unconditionally remove the check for Writable in getClass. My thought is:
* Users can always enforce the constraint later if they want to.
* All uses of the method with Hadoop's code base don't want to limit the output.
* The check isn't consistent. (It is only applied for aliases, not natural class names.)
* Removing the check won't break any callers since they couldn't get non-Writables before.

Reply:
Sounds good, thanks for looking!

@ayushtkn if you're ok with that I can just revert my last commit

Reply:
Go ahead

Reply:
@ayushtkn @omalley -- ok we're back to the original change here. Please give it another look and let me know if you'd like me to make any other changes.

Thanks again